x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/read_rules_route.ts

```diff
@@ -41,8 +44,24 @@ export const createReadRulesRoute: Hapi.ServerRoute = {
         id,
         ruleId,
       });
+      const ruleStatuses = await savedObjectsClient.find({
```

Comment:
You want to move the above code because rule _could_ be `null` right in this check like so:

```ts
      const rule = await readRules({
        alertsClient,
        id,
        ruleId,
      });
      if (rule != null) {
        const ruleStatuses = await savedObjectsClient.find({
          type: ruleStatusSavedObjectType,
          perPage: 10,
          search: `"${rule.id}"`,
          searchFields: ['alertId'],
        });
        ruleStatuses.saved_objects.sort((a, b) => {
          const dateA = new Date(a.attributes.statusDate);
          const dateB = new Date(b.attributes.statusDate);
          if (dateA < dateB) {
            return 1;
          } else if (dateA === dateB) {
            return 0;
          }
          return -1;
        });
        return transformOrError(rule, ruleStatuses); // update this to run with an array of rule statuses
      } else {
        return getIdError({ id, ruleId });
      }
    } catch (err) {
      return transformError(err);
    }
```